atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java

```diff
@@ -440,7 +445,24 @@ private TransactionManager serializableInternal(@Output List<AutoCloseable> clos
         TransactionKeyValueServiceManager transactionKeyValueServiceManager = initializeCloseable(
                 () -> {
                     TransactionKeyValueServiceManager tkvsm;
-                    tkvsm = new DefaultTransactionKeyValueServiceManager(keyValueService);
+                    if (config().transactionKeyValueService().isPresent()) {
+                        TransactionKeyValueServiceManagerFactory<?> tkvsmfFactory =
```

Comment:
I think for now I am going to ignore the hard question of "what happens if someone started using kvs-migrator and then their config broke... but it's a legitimate question and we probably should eventually handle this.

Reply:
Yeah agree with not creating state until actually configured (would have pushed strongly for it if we weren't doing that!)

For the migration question: agreed, that's probably the subject of its own internal RFC...